lib/CamDiag.py

```diff
@@ -386,19 +426,29 @@ def create_climo(self, baseline=False):
                 if not os.path.exists(avg_script_path):
                     msg = "Time averaging file '{}' is missing. Script is ending here.".format(avg_script_path)
                     end_diag_script(msg)
-
-                #Create averaging script import statement:
-                avg_func_import_statement = "from {} import {}".format(avg_func_name, avg_func_name)
-
-                #Run averaging script import statement:
-                exec(avg_func_import_statement)
-
-                #Create actual function call:
-                avg_func = avg_func_name+'(case_name, input_ts_loc, output_loc, var_list, overwrite_climo)'
-
-                #Evaluate (run) averaging script function:
-                eval(avg_func)
-
+                if avg_script_path not in sys.path:
+                    print(f"[INFO] Inserting to sys.path: {avg_script_path}")
+                    sys.path.insert(0, avg_script_path)
+                # NOTE: when we move to making this into a proper package, this path-checking stuff should be removed and dealt with on the package-level.
+                    
+                # Arguments; check if user has specified custom arguments
+                avg_func_args = [case_name, input_ts_loc, output_loc, var_list]
+                avg_func_kwargs = {"clobber":overwrite_climo}
+                if has_opt:
+                    if ('args' in opt):
+                        # RULES: it has to be a list of strings, and then we will take whatever of those are in locals
+                        assert isinstance(opt['args'], list)
```

Comment:
Can you add an error message like so:

`assert isinstance(opt['args'], list), "Function arguments must be of type list."`